packages/nimble-components/src/number-field/tests/number-field-matrix.stories.ts

```diff
@@ -76,6 +80,17 @@ export const numberFieldThemeMatrix: StoryFn = createMatrixThemeStory(
     ])
 );
 
+export const numberFieldInteractionsThemeMatrix: StoryFn = createMatrixThemeStory(
+    createMatrix(component, [
+        interactionStates.filter(x => x[0] && !x[0].includes('Active')),
+        disabledStates,
+        hideStepStates.slice(0, 1),
+        valueStates.slice(1, 2),
+        errorStates.slice(0, 2),
```

Comment:
Similar feedback here -- can we come up with a better approach than calling `slice` to pick specific states that we want to test. Some of the downsides I see to this approach are that it is very difficult to look at this code and understand what set of states you are trying to tests and that this code becomes unnecessarily tied to the order in which states are added to arrays.

I think the `filter` approach you take in the toggle button tests would be more readable, especially with a comment saying something like _only include states where XYZ_ or _skip states with XYZ_.